### PR TITLE
Handle IncompleteRead in the composer when on Python 3.

### LINKED ISSUE #2758: IncompleteRead while Composer polls for repomd files
It seems that Python's http library can raise a new Exception in Python 3 that we don't catch:

```
[2018-11-20 14:42:34][fedmsg.consumers    INFO]epel7-testing Polling http://download01.phx2.fedoraproject.org/pub/epel/testing/7/aarch64/repodata/repomd.xml
[2018-11-20 14:42:34][fedmsg.consumers   ERROR]epel7-testing Exception in ComposerThread(epel7-testing)
Traceback (most recent call last):
  File "/usr/lib/python3.7/site-packages/bodhi/server/consumers/masher.py", line 410, in work
    self._compose_updates()
  File "/usr/lib/python3.7/site-packages/bodhi/server/consumers/masher.py", line 959, in _compose_updates
    self._wait_for_sync()
  File "/usr/lib/python3.7/site-packages/bodhi/server/consumers/masher.py", line 1327, in _wait_for_sync
    newsum = hashlib.sha1(masterrepomd.read()).hexdigest()
  File "/usr/lib64/python3.7/http/client.py", line 460, in read
    s = self._safe_read(self.length)
  File "/usr/lib64/python3.7/http/client.py", line 612, in _safe_read
    raise IncompleteRead(b''.join(s), amt)
http.client.IncompleteRead: IncompleteRead(3295 bytes read, 1 more expected)
```

This one should be pretty easy to fix - we just need to catch this new Exception and retry, similar to what we do if a URLError is raised.

### PR DESCRIPTION
**Pull request by bowlofeggs**
fixes #2758

Signed-off-by: Randy Barlow <randy@electronsweatshop.com>

### CODE REVIEW

**Review by puiterwijk — APPROVED** [review]
LGTM

**Review by Zlopez — APPROVED** [review]
LGTM

File: bodhi/tests/server/consumers/test_masher.py [hunk 1]
```diff
@@ -25,6 +25,11 @@
 import tempfile
 import time
 import unittest
+try:
+    from http.client import IncompleteRead
+except ImportError:
+    # Python 2 does not have this Exception.
+    IncompleteRead = None
```
Comment (pypingou) [inline bodhi/tests/server/consumers/test_masher.py:32]: You could import this directly in the test to save you from catching the ImportError

**Comment by pypingou** [comment]
Looks fine to me

File: bodhi/tests/server/consumers/test_masher.py (hunk 1 above)
Comment (bowlofeggs) [reply bodhi/tests/server/consumers/test_masher.py:32]: That would violate PEP-8, so I prefer this style. Bodhi will [drop Python 2](https://github.com/fedora-infra/bodhi/issues/2759) soon anyway, so this code won't be here for long.

**Pull request merged**